targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_K66F/serial_api.c

```diff
@@ -340,14 +339,55 @@ const PinMap *serial_rts_pinmap()
     return PinMap_UART_RTS;
 }
 
-void serial_wait_tx_complete(uint32_t uart_index)
+static int serial_is_enabled(uint32_t uart_index)
+{
+    int clock_enabled = 0;
+    switch (uart_index) {
+        case 0:
+            clock_enabled = (SIM->SCGC4 & SIM_SCGC4_UART0_MASK) >> SIM_SCGC4_UART0_SHIFT;
```

Comment:
Personally, I'd be inclined to pare this down to more like:

    static bool serial_is_enabled(uint32_t uart_index)
    {
        switch (uart_index) {
            case 0:
                return SIM->SCGC4 & SIM_SCGC4_UART0_MASK;
            default:
                return false;
        }
    }

Let `bool` take the strain. But this is fine.